File: Antegral.java

public class Antegral {

    private IFunc mFunc;
    private boolean debug = true;
    private double mTol = 0.001;

    public Antegral(IFunc func) {
        mFunc = func;
    }

    public Antegral(IFunc func, double tol) {
        mFunc = func;
        mTol = tol;
    }

    public double calcute(double start, double end) {

        double p = start;

        return 5;
    }

    private void Log(double val) {
        System.out.println(val);
    }
}

File: js/Antegral.js
var mTol = 0.00001;

exports.calcute = function (start,end, func) {

    var p = start;
    

    var sum = 0;
    while (p < end) {
        p += mTol;
        Log("p", p);
        sum += func(p);
        Log("sum", sum);
    }
    sum *= mTol;
    return sum;
}
var debug = false;
function Log(val) {
    if (debug)
        console.log(val);
}